applications/nrf_desktop/CMakeLists.txt

```diff
@@ -6,26 +6,24 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-
-if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}")
-  message(FATAL_ERROR
-          "Board ${BOARD} is not supported.\n"
-          "Please make sure board specific configuration files are added to "
-          "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}")
-endif()
-
 ################################################################################
 
 # The application uses the configuration/<board> scheme for configuration files.
-set(APPLICATION_CONFIG_DIR "configuration/\${BOARD}")
-
+set(APPLICATION_CONFIG_DIR "${CMAKE_CURRENT_SOURCE_DIR}/configuration/\${BOARD}")
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project("nRF52 Desktop"
         VERSION 0.1)
 
 ################################################################################
 
+if (NOT EXISTS "${APPLICATION_CONFIG_DIR}")
```

Comment:
We already pass the `APPLICATION_CONFIG_DIR` to Zephyr build system before the actual check. Can the fatal error be reported there in this case?

Reply:
Also from what I understood we expect that Zephyr build system may modify content of the `APPLICATION_CONFIG_DIR` (that's why the check was moved). Consider logging the actually used configuration directory (`APPLICATION_CONFIG_DIR`) in that case.

Reply:
It already does, e.g.: `Merged configuration '/tmp/bb/nrf/applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf'`